sdk/eventhub/azure-eventhubs/azure/eventhub/eventprocessor/partition_processor.py

```diff
@@ -0,0 +1,47 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# -----------------------------------------------------------------------------------
+
+from typing import List
+from abc import ABC, abstractmethod
+from enum import Enum
+from .checkpoint_manager import CheckpointManager
+
+from azure.eventhub import EventData
+
+
+class CloseReason(Enum):
+    SHUTDOWN = 0  # user call EventProcessor.stop()
+    OWNERSHIP_LOST = 1  # lose the ownership of a partition.
+    EVENTHUB_EXCEPTION = 2  # Exception happens during receiving events
+
+
+class PartitionProcessor(ABC):
+    def __init__(self, eventhub_name, consumer_group_name, partition_id, checkpoint_manager: CheckpointManager):
+        self._partition_id = partition_id
+        self._eventhub_name = eventhub_name
+        self._consumer_group_name = consumer_group_name
+        self._checkpoint_manager = checkpoint_manager
+
+    async def close(self, reason):
+        """Called when EventProcessor stops processing this PartitionProcessor.
+
+        There are different reasons to trigger the PartitionProcessor to close.
+        Refer to enum class CloseReason
+
+        """
+        pass
+
+    @abstractmethod
+    async def process_events(self, events: List[EventData]):
```

Comment:
I would prefer to get a `checkpoint` callable argument (or a checkpoint manager) as an argument to the method. Same for close and (potentially) process_error.

Reply:
The idea is since all methods may need to use checkpoint manager, make it a class member variable.

Reply:
This has been concluded that we continue to use a class